src/Plugin/Queue/Handlers/Source/CreateKafka.php

```diff
@@ -0,0 +1,342 @@
+
+			$request = $manticoreClient->sendRequest($query);
+			if ($request->hasError()) {
+				throw ManticoreSearchClientError::create((string)$request->getError());
+			}
+
+			$escapedPayload = str_replace("'", "\\'", $payload->originQuery);
+			/** @l $query */
+			$query = /** @lang ManticoreSearch */
+				'INSERT INTO ' . Payload::SOURCE_TABLE_NAME .
+				' (id, type, name, full_name, buffer_table, attrs, original_query) VALUES ' .
+				"(0, '" . self::SOURCE_TYPE_KAFKA . "', '$options->name','{$options->name}_$i'," .
+				"'_buffer_{$options->name}_$i', '$attrs', '$escapedPayload')";
+
+			$request = $manticoreClient->sendRequest($query);
+			if ($request->hasError()) {
+				throw ManticoreSearchClientError::create((string)$request->getError());
+			}
+		}
+
+		self::handleOrphanViews($options->name, $options->numConsumers, $manticoreClient);
+
+		return TaskResult::none();
+	}
+
+	/**
+	 * @throws ManticoreSearchClientError
+	 */
+	public static function handleOrphanViews(string $sourceName, int $maxIndex, Client $client): void {
+		$viewsTable = Payload::VIEWS_TABLE_NAME;
+		if (!$client->hasTable($viewsTable)) {
+			return;
+		}
+
+		$sql = /** @lang Manticore */
+			"SELECT * FROM $viewsTable " .
+			"WHERE MATCH('@source_name \"" . $sourceName . "_*\"') AND suspended=1";
+
+		$request = $client->sendRequest($sql);
+		if ($request->hasError()) {
+			throw ManticoreSearchClientError::create((string)$request->getError());
+		}
+
+		$result = $request->getResult();
+
+		if (!is_array($result[0])) {
+			return;
+		}
+
+		$views = $result[0]['data'];
+		$viewsCount = sizeof($views);
+
+
+		if ($viewsCount > 0 && $viewsCount < $maxIndex) {
+			$originalQuery = (string)$views[0]['original_query'];
+			$viewName = (string)$views[0]['name'];
+
+			$parser = new PHPSQLParser($originalQuery);
+			$destinationTableName = $parser->parsed['VIEW']['to']['no_quotes']['parts'][0];
+			unset($parser->parsed['CREATE'], $parser->parsed['VIEW']);
+
+			CreateViewHandler::createViewRecords(
+				$client, $viewName, $parser->parsed,
+				$sourceName, $originalQuery, $destinationTableName,
+				$maxIndex, $viewsCount, 1
+			);
+		} else {
+			$ids = self::getOrphanIds($views, $sourceName, $maxIndex);
+
+			if ($ids === []) {
+				return;
+			}
+
+			$ids = implode(',', $ids);
+			Buddy::debug("Remove orphan views records ids ($ids)");
+			$sql = /** @lang manticore */
+				"DELETE FROM $viewsTable WHERE id in ($ids)";
+			$rawResult = $client->sendRequest($sql);
+			if ($rawResult->hasError()) {
+				throw ManticoreSearchClientError::create((string)$rawResult->getError());
+			}
+		}
+	}
+
+	/**
+	 * @param array<int, array<string>> $views
+	 * @param string $sourceName
+	 * @param int $maxIndex
+	 * @return array<string>
+	 */
+	public static function getOrphanIds(array $views, string $sourceName, int $maxIndex): array {
+		$ids = [];
+		foreach ($views as $orphanView) {
+			$viewSourceName = explode('_', $orphanView['source_name']);
```

Comment:
I think explode here is overkill, probably it's a good to use substr kind of that will be faster without overhead of rebuilding array

```php
$viewSourceName = $orphanView['source_name'];
$lastUnderscorePos = strrpos($viewSourceName, '_');
if ($lastUnderscorePos !== false) {
    $viewSourceName = substr($viewSourceName, 0, $lastUnderscorePos);
}
```